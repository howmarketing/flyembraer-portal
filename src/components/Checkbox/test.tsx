import { render, screen, waitFor } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import CheckBox from "."

describe("<CheckBox />", () => {
	it("should render the CheckBox and match the previous version", () => {
		const { container } = render(<CheckBox id="label">Label</CheckBox>)

		expect(screen.queryByLabelText("Label")).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it("should be checked by default", () => {
		render(
			<CheckBox id="label" defaultChecked>
				Label
			</CheckBox>
		)

		expect(screen.queryByLabelText("Label")).toBeChecked()
	})

	it("should change the checked state of the CheckBox", async () => {
		render(<CheckBox id="label">Label</CheckBox>)

		expect(screen.getByRole("checkbox")).not.toBeChecked()

		userEvent.click(screen.getByRole("checkbox"))

		await waitFor(() => {
			expect(screen.getByRole("checkbox")).toBeChecked()
		})
	})

	it("should show the field is required", () => {
		render(
			<CheckBox id="label" required>
				Label
			</CheckBox>
		)

		expect(screen.queryByLabelText("Label *")).toBeInTheDocument()
	})

	it("should be accessible by tab", () => {
		render(<CheckBox id="label">Label</CheckBox>)

		const input = screen.getByLabelText("Label")
		expect(document.body).toHaveFocus()

		userEvent.tab()
		expect(input).toHaveFocus()
	})
})
