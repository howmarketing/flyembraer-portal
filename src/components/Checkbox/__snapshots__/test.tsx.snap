// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should render the CheckBox and match the previous version 1`] = `
<div
  class="c-lctIFO"
>
  <input
    aria-checked="false"
    aria-required="false"
    class="c-chkGgc"
    id="label"
    name="label"
    type="checkbox"
  />
  <label
    class="c-kSThZz"
    for="label"
  >
    Label
  </label>
</div>
`;
